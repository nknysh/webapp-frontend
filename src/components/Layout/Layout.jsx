import React, { useMemo } from 'react';
import { withRouter } from 'react-router-dom';
import { compose, prop, pick, includes } from 'ramda';
import { SnackbarProvider } from 'notistack';
import { DriftWidget } from '@pure-escapes/webapp-ui-components';

import { DRIFT_APP_ID, DRIFT_ENABLED_ROLES } from 'config';
import Notifications from 'components/Notifications';
import { withUser } from 'hoc';

import { propTypes } from './Layout.props';
import { StyledLayout, LayoutChildren, LayoutHeader, LayoutFooter } from './Layout.styles';

const ANCHOR_ORIGIN = Object.freeze({
  vertical: 'top',
  horizontal: 'right',
});

/**
 * Drift enabled
 *
 * Compares role to env var to see if drift should be enabled
 *
 * @param {string} role
 * @returns {boolean}
 */
const driftEnabled = role => includes('all', DRIFT_ENABLED_ROLES) || includes(role, DRIFT_ENABLED_ROLES);

export const Layout = ({ user, children, location: { pathname } }) => {
  const enableDrift = useMemo(() => driftEnabled(prop('type', user)), [user]);
  const driftAttributes = useMemo(
    () => pick(['email', 'title', 'firstName', 'lastName', 'phoneNumber', 'mobileNumber'], user || {}),
    [user]
  );
  const userUuid = useMemo(() => prop('uuid', user), [user]);

  return (
    <React.Fragment>
      <StyledLayout>
        <SnackbarProvider anchorOrigin={ANCHOR_ORIGIN}>
          <Notifications />
        </SnackbarProvider>
        <DriftWidget appId={DRIFT_APP_ID} attributes={driftAttributes} enabled={enableDrift} userId={userUuid} />
        <LayoutHeader currentPath={pathname} />
        <LayoutChildren>{children}</LayoutChildren>
        <LayoutFooter currentPath={pathname} />
      </StyledLayout>
    </React.Fragment>
  );
};

Layout.propTypes = propTypes;

export default compose(
  withRouter,
  withUser
)(Layout);
