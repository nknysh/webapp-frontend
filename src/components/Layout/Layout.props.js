import PropTypes from 'prop-types';

export const propTypes = {
  isAppVersionDeprecated: PropTypes.bool,
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node]),
};
