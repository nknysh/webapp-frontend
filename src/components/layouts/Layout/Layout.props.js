import PropTypes from 'prop-types';

export const propTypes = {
  children: PropTypes.oneOfType([PropTypes.element, PropTypes.node]),
};
