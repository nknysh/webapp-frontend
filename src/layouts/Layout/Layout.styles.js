import styled from 'styled-components';

import { Header, Footer } from 'containers';

export const StyledLayout = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100%;
`;

export const LayoutHeader = styled(Header)``;
export const LayoutFooter = styled(Footer)`
  flex: 0 1;
`;

export const LayoutChildren = styled.div`
  flex: 1 1 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
`;

export const LayoutContent = styled.div`
  flex: 1;
`;
