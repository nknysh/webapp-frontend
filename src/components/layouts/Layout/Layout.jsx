import React from 'react';
import { Helmet } from 'react-helmet';
import { withRouter } from 'react-router-dom';
import { prop, compose } from 'ramda';
import { SnackbarProvider } from 'notistack';

import { Notifications } from 'components';

import config from 'config/ui';

import { propTypes } from './Layout.props';
import { StyledLayout, LayoutChildren, LayoutHeader, LayoutFooter } from './Layout.styles';

export const Layout = ({ children, location: { pathname } }) => (
  <React.Fragment>
    <Helmet>
      <title>{prop('title', config)}</title>
    </Helmet>
    <StyledLayout>
      <SnackbarProvider
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right',
        }}
      >
        <Notifications />
      </SnackbarProvider>
      <LayoutHeader currentPath={pathname} />
      <LayoutChildren>{children}</LayoutChildren>
      <LayoutFooter currentPath={pathname} />
    </StyledLayout>
  </React.Fragment>
);

Layout.propTypes = propTypes;

export default compose(withRouter)(Layout);
