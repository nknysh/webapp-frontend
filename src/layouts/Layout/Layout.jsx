import React from 'react';

import { propTypes } from './Layout.props';
import { StyledLayout, LayoutChildren, LayoutHeader, LayoutFooter } from './Layout.styles';

export const Layout = ({ children }) => (
  <StyledLayout>
    <LayoutHeader />
    <LayoutChildren>{children}</LayoutChildren>
    <LayoutFooter />
  </StyledLayout>
);

Layout.propTypes = propTypes;

export default React.memo(Layout);
