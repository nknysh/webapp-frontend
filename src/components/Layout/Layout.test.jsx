import React from 'react';

import { Layout } from './Layout';

const getComponent = props =>
  shallow(
    <Layout location={{ pathname: '' }} {...props}>
      <h1>Some content</h1>
    </Layout>
  );

describe.skip('<Layout />', () => {
  describe('render', () => {
    it('matches snapshot', () => {
      expect(getComponent()).toMatchSnapshot();
    });
  });
});
