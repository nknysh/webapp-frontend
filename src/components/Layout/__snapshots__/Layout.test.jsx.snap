// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> render matches snapshot 1`] = `
<Fragment>
  <Layoutstyles__StyledLayout>
    <SnackbarProvider
      TransitionComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "propTypes": Object {
            "children": [Function],
            "direction": [Function],
            "in": [Function],
            "onEnter": [Function],
            "onEntering": [Function],
            "onExit": [Function],
            "onExited": [Function],
            "style": [Function],
            "timeout": [Function],
          },
          "render": [Function],
        }
      }
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      autoHideDuration={5000}
      classes={Object {}}
      dense={false}
      hideIconVariant={false}
      iconVariant={Object {}}
      maxSnack={3}
      preventDuplicate={false}
    >
      <Memo(WithSnackbar(withNotifications(Notifications))) />
    </SnackbarProvider>
    <Memo(oc)
      attributes={Object {}}
      enabled={true}
    />
    <Layoutstyles__LayoutHeader
      currentPath=""
    />
    <Layoutstyles__LayoutChildren>
      <h1>
        Some content
      </h1>
    </Layoutstyles__LayoutChildren>
    <Layoutstyles__LayoutFooter
      currentPath=""
    />
  </Layoutstyles__StyledLayout>
</Fragment>
`;
